BootloaderCorePkg/Stage1B/Stage1B.c

```diff
@@ -31,6 +34,16 @@ LoadComponentCallback (
   default:
     break;
   }
+
+  //Check the boot mode
+  BootMode = GetBootMode();

```

Comment:
I think you should define a new function for this.  And only call this with  PROGESS_ID_AUTHENTICATE case.